cmd/metrics-v2.go

```diff
@@ -1265,9 +1265,13 @@ func (c *minioClusterCollector) Collect(out chan<- prometheus.Metric) {
 		}
 	}
 
+	peerDeadlineCtx, pCancel := context.WithTimeout(GlobalContext, 1*time.Minute)
+	defer pCancel()
+	reportDeadlineCtx, rCancel := context.WithTimeout(GlobalContext, 1*time.Minute)
+	defer rCancel()
 	// Call peer api to fetch metrics
-	peerCh := globalNotificationSys.GetClusterMetrics(GlobalContext)
-	selfCh := ReportMetrics(GlobalContext, GetAllGenerators)
+	peerCh := globalNotificationSys.GetClusterMetrics(peerDeadlineCtx)
+	selfCh := ReportMetrics(reportDeadlineCtx, GetAllGenerators)
```

Comment:
This should generally match the Prometheus deadline itself for each scrape replies - @poornas - perhaps should be configurable. 

IMO we should cache all our Prometheus values and update them once in 30secs to avoid aggressive prometheus scrape intervals.  Allowing lesser load on the system.

Reply:
The problem with caching is that if some nodes go down, we would still serve stale cached entries - just updating the cache is not enough, it needs to be cleared and rebuilt to be correct

Reply:
Correct but prometheus will not know if the node is down instantaneously @poornas if our cache window is smaller than Prometheus scrape window we are good.

Reply:
Mainly thing is querying disk space utilization etc are not worth doing everything as a fresh call to the drives. Nothing much would have changed in 15-30secs

Reply:
that is true, will take another stab at it